modules/generator/registry/gauge.go

```diff
@@ -118,31 +118,27 @@ func (g *gauge) collectMetrics(appender storage.Appender, timeMs int64, external
 
 	activeSeries = len(g.series)
 
-	labelsCount := 0
-	if activeSeries > 0 && g.series[0] != nil {
-		labelsCount = len(g.series[0].labels.names)
-	}
-	lbls := make(labels.Labels, 1+len(externalLabels)+labelsCount)
-	lb := labels.NewBuilder(lbls)
-
-	// set metric name
-	lb.Set(labels.MetricName, g.metricName)
-	// set external labels
-	for name, value := range externalLabels {
-		lb.Set(name, value)
-	}
-
 	for _, s := range g.series {
 		t := time.UnixMilli(timeMs)
+		lbls := make(labels.Labels, 1+len(externalLabels)+len(s.labels.names))
```

Comment:
We might want to keep the allocations outside of the loop.  I think I saw a `Reset()` method on the lb object that might empty this out.

Reply:
good call